nixos.lib2geom build failure on i686

### Describe the bug
A clear and concise description of what the bug is.

### Steps To Reproduce
Steps to reproduce the behavior:
1.  nixos-rebuild switch 
or
2. nix build nixos.lib2geom


### Expected behavior
package should build/install

### Additional context
Add any other context about the problem here.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
nix-shell -p nix-info --run "nix-info -m"
 - system: `"i686-linux"`
 - host os: `Linux 5.12.19, NixOS, 21.05.2031.4feab93d28e (Okapi)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.11pre307874.740ecbc1627"`
 - channels(sss): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```


```
nix build nixos.lib2geom        
error: build of '/nix/store/avhl27jpvr6kjayzrpgwyfx0xv39w40n-lib2geom-1.1.drv' on 'ssh://nix-builder@192.168.0.2' failed: builder for '/nix/store/avhl27jpvr6kjayzrpgwyfx0xv39w40n-lib2geom-1.1.drv' failed with exit code 1
builder for '/nix/store/avhl27jpvr6kjayzrpgwyfx0xv39w40n-lib2geom-1.1.drv' failed with exit code 1; last 10 log lines:
  Total Test time (real) =   0.68 sec
  
  The following tests FAILED:
            3 - bezier-test (Failed)
            4 - choose-test (Failed)
  Errors while running CTest
  FAILED: CMakeFiles/test.util
  cd /tmp/nix-build-lib2geom-1.1.drv-0/source/build && /nix/store/yyrj8iz83yr2nqba65ys1xy2f1rsjn9w-cmake-3.19.7/bin/ctest --force-new-ctest-process
  ninja: build stopped: subcommand failed.
  builder for '/nix/store/avhl27jpvr6kjayzrpgwyfx0xv39w40n-lib2geom-1.1.drv' failed with exit code 1
[0 built (1 failed)]
error: build of '/nix/store/avhl27jpvr6kjayzrpgwyfx0xv39w40n-lib2geom-1.1.drv' failed
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/by-name/li/lib2geom/package.nix
+++ b/pkgs/by-name/li/lib2geom/package.nix
@@ -57,22 +57,25 @@ stdenv.mkDerivation rec {
   checkPhase =
     let
       disabledTests =
-        lib.optionals stdenv.hostPlatform.isAarch64 [
-          # Broken on all platforms, test just accidentally passes on some.
-          # https://gitlab.com/inkscape/lib2geom/-/issues/63
-          "elliptical-arc-test"
-        ]
-        ++ lib.optionals stdenv.hostPlatform.isMusl [
+        lib.optionals stdenv.hostPlatform.isMusl [
           # Fails due to rounding differences
           # https://gitlab.com/inkscape/lib2geom/-/issues/70
           "circle-test"
         ]
         ++ lib.optionals (stdenv.hostPlatform.system != "x86_64-linux") [
+          # Broken on all platforms, test just accidentally passes on some.
+          # https://gitlab.com/inkscape/lib2geom/-/issues/63
+          "elliptical-arc-test"
+
           # https://gitlab.com/inkscape/lib2geom/-/issues/69
           "polynomial-test"
 
           # https://gitlab.com/inkscape/lib2geom/-/issues/75
           "line-test"
+
+          # Failure observed on i686
+          "angle-test"
+          "self-intersections-test"
 
           # Failure observed on aarch64-darwin
           "bezier-test"

